ReSharper.FSharp/src/FSharp/FSharp.Psi.Intentions/src/Intentions/FunctionAnnotationAction.fs

```diff
@@ -44,6 +44,7 @@ module SpecifyTypes =
             let pattern = pattern.IgnoreInnerParens()
             match pattern with
             | :? ITypedPat | :? IUnitPat -> true
+            | :? IAttribPat as attribPat -> isAnnotated isTopLevel attribPat.Pattern
```

Comment:
Just in case, have you checked all other kinds of patterns? Doesn't something like `?a` have a similar structure?